bindings/py/cpp_src/bindings/algorithms/py_TemporalMemory.cpp

```diff
@@ -148,38 +238,86 @@ using namespace nupic::algorithms::connections;
+still valid.
+
+Argument learn
+    If true, segment activations will be recorded. This information is
+    used during segment cleanup.
+
+Argument extraActive
+    (optional) SDR of active external predictive inputs.
+
+Argument extraWinners
+    (optional) SDR of winning external predictive inputs.  When learning, only
+    these inputs are considered active.
+    ExtraWinners must be a subset of extraActive.
+
+See TM.compute() for details of the parameters.)",
+            py::arg("learn"),
+            py::arg("extraActive"),
+            py::arg("extraWinners"));
 
         py_HTM.def("getPredictiveCells", [](const HTM_t& self)
-            { return self.getPredictiveCells();});
+            { return self.getPredictiveCells();},
+R"()");
 
         py_HTM.def("getWinnerCells", [](const HTM_t& self)
         {
-            auto winnerCells = self.getWinnerCells();
-
-            return py::array_t<nupic::UInt32>(winnerCells.size(), winnerCells.data());
-        });
+            auto dims = self.getColumnDimensions();
+            dims.push_back( self.getCellsPerColumn() );
+            SDR *winnerCells = new SDR( dims );
+            self.getWinnerCells(*winnerCells);
+            return winnerCells;
+        },
+R"()");
 
         py_HTM.def("getActiveSegments", [](const HTM_t& self)
-            { return self.getActiveSegments(); });
+            { return self.getActiveSegments(); },
+R"()");
 
         py_HTM.def("getMatchingSegments", [](const HTM_t& self)
-            { return self.getMatchingSegments(); });
+            { return self.getMatchingSegments(); },
+R"()");
 
         py_HTM.def("cellsForColumn", [](HTM_t& self, UInt columnIdx)
         {
             auto cells = self.cellsForColumn(columnIdx);
 
             return py::array_t<nupic::UInt32>(cells.size(), cells.data());
-        });
+        },
+R"(Returns list of indices of cells that belong to a mini-column.
+
+Argument column is sparse index of a mini-column.)");
+
+        py_HTM.def("createSegment", &HTM_t::createSegment,
+R"(Create a segment on the specified cell. This method calls
+createSegment on the underlying connections, and it does some extra
+bookkeeping. Unit tests should call this method, and not
+connections.createSegment().
+
+Argument cell
+    Index of Cell to add a segment to.
 
-        py_HTM.def("createSegment", &HTM_t::createSegment);
+Returns the created segment (index handle).)");
 
-        py_HTM.def("numberOfCells",   &HTM_t::numberOfCells);
+        py_HTM.def("numberOfCells",   &HTM_t::numberOfCells,
+R"(Returns the number of cells in this TemporalMemory.)");
 
-        py_HTM.def("numberOfColumns", &HTM_t::numberOfColumns);
+        py_HTM.def("numberOfColumns", &HTM_t::numberOfColumns,
+R"(Returns the total number of mini-columns.)");
 
-        py_HTM.def_property_readonly("extra", [](const HTM_t &self) { return self.extra; } );
+        py_HTM.def_property_readonly("extra", [](const HTM_t &self) { return self.extra; },
+R"()");
```

Comment:
Extra would deserve an explanation

Reply:
Yes. I intend to change the variable name in both c++ and python to externalPredictive, and then add documentation.